### DISCUSSION TITLE
Manually setting values on demand is not working.

### DISCUSSION THREAD

**Discussion by vanor89** [react-hook-form/react-hook-form#2356] [Q&A]
I have a form that has dropdown menus which contents depend of the selection of their previous dropdowns, for example
If in the first dropdown I select `a` then the second dropdown has options `d` and `e` and if I select in the first dropdown `b` then the second dropdown has options `c` and `f`.

watch and set value are not working as expected, I expected to be able to have a init function that ran first in my components lifecycle, in this function I call `setValue` for one of the forms value and then immediately call 'watch' for that value, it returns undefined
```
    setValue('value', varWithValue);
    console.log(watch('curriculum')); <------ undefined, not value within varWithValue
```

**Comment by bluebill1049** [comment]
provide a reproduceable codesandbo.
